v22.2/trigram-indexes.md

```diff
@@ -0,0 +1,354 @@
+  ~~~
+
+  ~~~ sql
+  CREATE INDEX {optional name} ON {table} USING GIST({column} gist_trgm_ops);
+  ~~~
+
+  {{site.data.alerts.callout_info}}
+  GIN and GiST indexes are implemented identically on CockroachDB. `GIN` and `GIST` are therefore synonymous when defining a trigram index.
+  {{site.data.alerts.end}}
+
+- Using `CREATE INVERTED INDEX`:
+
+  ~~~ sql
+  CREATE INVERTED INDEX {optional name} ON {table} ({column} {opclass});
+  ~~~
+
+### Comparisons
+
+Trigram indexes on `STRING` columns support the following comparison operators:
+
+- **equality**: [`=`](functions-and-operators.html#operators). Note that standard [`btree` secondary indexes](create-index.html#parameters) may perform better than trigram indexes for equality searches.
+- **pattern matching (case-sensitive)**: [`LIKE`](functions-and-operators.html#operators)
+- **pattern matching (case-insensitive)**: [`ILIKE`](functions-and-operators.html#operators)
+- **similarity matching**: [`%`](functions-and-operators.html#operators). This operator returns `true` if the strings in the comparison have a similarity that meets or exceeds the threshold set by the `pg_trgm.similarity_threshold` [session variable](show-vars.html#pg_trgm_similarity_threshold).
+
+For usage examples, see [Use a trigram index to speed up fuzzy string matching](#use-a-trigram-index-to-speed-up-fuzzy-string-matching).
+
+## Examples
+
+### Simple examples
+
+Suppose you have a table with the following columns:
+
+~~~ sql
+CREATE TABLE t (a INT, w STRING);
+~~~
+
+The following examples illustrate how to create various trigram indexes on column `w`.
+
+A GIN index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~sql
+CREATE INDEX ON t USING GIN (w gin_trgm_ops);
+~~~
+
+A partial index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN (w gin_trgm_ops) WHERE a > 0;
+~~~
+
+A multi-column index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN (a, w gin_trgm_ops);
+~~~
+
+An expression index with trigram matching enabled:
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN ((lower(w)) gin_trgm_ops);
+~~~
+
+### Use a trigram index to speed up fuzzy string matching
+
+Create a table with a `STRING` column:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE t (w STRING);
+~~~
+
+Populate the table with sample values:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+INSERT INTO t VALUES
+  ('foo'),
+  ('bar'),
+  ('wordy'),
+  ('world'),
+  ('whorl'),
+  ('wort'),
+  ('worm'),
+  ('norm'),
+  ('weird'),
+  ('worried'),
+  ('wofoord'),
+  ('wobarrd');
+
+INSERT INTO t SELECT 'empty' FROM generate_series(1, 10000);
+~~~
+
+Retrieve the columns with values similar to `word`, using the `%` operator. Sort the results by the output of the `similarity()` [built-in function](functions-and-operators.html#trigrams-functions):
```

Comment:
nit: somewhere before running the `SELECT` query **without** the trigram function it might be clearer to mention that. As in, "First, retrieve the columns with values similar.... without using a trigram index." I feel like this will help reinforce that we're kind of evaluating the difference in performance between the two to begin.

Reply:
Great point; I clarified this.